picard/util/__init__.py

```diff
@@ -391,13 +391,10 @@ def album_artist_from_path(filename, album, artist):
     """
     if not album:
         dirs = os.path.dirname(filename).replace('\\', '/').lstrip('/').split('/')
-        if len(dirs) == 0:
-            return album, artist
         # Strip disc subdirectory from list
-        if len(dirs) > 0:
-            if re.search(r'(^|\s)(CD|DVD|Disc)\s*\d+(\s|$)', dirs[-1], re.I):
-                del dirs[-1]
-        if len(dirs) > 0:
+        if re.search(r'\b(?:CD|DVD|Disc)\s*\d+\b', dirs[-1], re.I):
```

Comment:
Why was the check for `len(dirs) == 0` from above removed? `dirs[-1]` will fail in this case.

Reply:
As said in the commit message:
> len(dirs) == 0 cannot happen, because `whatever.split('/')` is always a list with at least 1 element.

So `dirs` contains at least one element, the empty string'

Reply:
This is actually tested in https://github.com/metabrainz/picard/pull/1493/files#diff-85c7da7d97a696e8017da5742a876475R211